vllm/model_executor/layers/tuned_gemm.py

```diff
@@ -89,7 +90,14 @@ def mm(self, inp, weights):
                 ]).drop_duplicates()
                 self.tuned_df.to_csv(self.untune_path, index=False)
 
-            if n == 1 and inp_view.dtype == torch.float16:
+            if ((n == 4 or n == 3 or n== 2 or n == 1)
+                    and inp_view.dtype == torch.float16):
+                out = torch.empty(inp_view.shape[0],
+                                  weights.shape[0],
+                                  dtype=inp_view.dtype,
+                                  device='cuda')
+                _custom_C.wvSpltK(weights, inp_view, out, n, self.CuCount)
+            elif n == 1 and inp_view.dtype == torch.float16:
```

Comment:
We are never going to reach here now, aren't we? Might as well remove the code path if we are certain that wvSpltK is always superior to LLMM1

Reply:
was missing k%8==0 condition. I left in LLMM1 because it was restrictive in passed, and we haven't extensively checked if there are cases where it outperforms.